src/components/maps/CameraViewer.jsx

```diff
@@ -0,0 +1,188 @@
+import Tiles from 'grommet/components/Tiles';
+import FormNextIcon from 'grommet/components/icons/base/FormNext';
+import FormPreviousIcon from 'grommet/components/icons/base/FormPrevious';
+import SpinningIcon from 'grommet/components/icons/Spinning';
+
+import {
+  MAPEXPLORER_INITIAL_STATE, MAPEXPLORER_PROPTYPES,
+  MAPEXPLORER_CAMERA_STATUS,
+} from '../../ducks/mapexplorer';
+
+const ITEMS_PER_PAGE = 6;
+const THUMBNAIL_SERVER_URL = 'https://thumbnails.zooniverse.org/';
+const THUMBNAIL_WIDTH = 320;
+const THUMBNAIL_HEIGHT = 200;
+
+class CameraViewer extends React.Component {
+  constructor(props) {
+    super(props);
+  
+    this.renderData = this.renderData.bind(this);
+    this.renderMetadata = this.renderMetadata.bind(this);
+    this.renderDataPaging = this.renderDataPaging.bind(this);
+    this.changeDataPaging = this.changeDataPaging.bind(this);
+  
+    this.state = {
+      page: 0,
+    };
+  }
+
+  componentDidMount() {
+    this.setState({
+      page: 0,
+    });
+  }
+  
+  render() {
+    if (!this.props.mapConfig) return null;
+    
+    return (
+      <Layer className="camera-viewer">
+        <Box className="content" align="center">
+          <Box className="camera-metadata">
+            {this.renderMetadata()}
+          </Box>
+          <Box className="camera-data">
+            {this.renderDataPaging()}
+            {this.renderData()}
+          </Box>
+          <Button label="Done" onClick={()=>{ Actions.mapexplorer.resetActiveCamera() }}></Button>
+        </Box>
+      </Layer>
+    );
+  }
+
+  renderMetadata() {
+    if (this.props.activeCameraMetadataStatus === MAPEXPLORER_CAMERA_STATUS.FETCHING) {
+      return <SpinningIcon />;
+    } else if (this.props.activeCameraMetadataStatus === MAPEXPLORER_CAMERA_STATUS.ERROR) {
+      return <Notification message='ERROR' status='critical' />
+    }
+    
+    if (this.props.activeCameraMetadataStatus === MAPEXPLORER_CAMERA_STATUS.SUCCESS &&
+        this.props.activeCameraMetadata) {
+
+      return (
+        <List>
+        {Object.keys(this.props.activeCameraMetadata).map(key => (
+          <ListItem
+            key={`camera-metadata-${key}`}
+            justify='between'
+            separator='horizontal'
+          >
+            <span>{key}</span>
+            <span>{this.props.activeCameraMetadata[key]}</span>
+          </ListItem>
+        ))}
+        </List>
+      );
+    }
+
+    return null;
+  }
+
+  renderData() {
+    if (this.props.activeCameraDataStatus === MAPEXPLORER_CAMERA_STATUS.FETCHING) {
+      return <SpinningIcon />;
+    } else if (this.props.activeCameraDataStatus === MAPEXPLORER_CAMERA_STATUS.ERROR) {
+      return <Notification message='ERROR' status='critical' />
+    }
+    
+    if (this.props.activeCameraDataStatus === MAPEXPLORER_CAMERA_STATUS.SUCCESS &&
+        this.props.activeCameraData) {
+      const data = this.props.activeCameraData;
+      const items = [];
+      for (let i = this.state.page * ITEMS_PER_PAGE;
```

Comment:
This loop totally works, but I might suggest considering a combination of `Array.slice()` to get the six data items you want and then `Array.map()` to iterate through for rendering:
```
const currentPageOfData = data.slice(startingIndex, ITEMS_PER_PAGE);
return (
  <Tiles>
    {currentPageOfData.map((currentItem, i) => {
      const fixedLocation = this.fixLocation(currentItem.location) // the regex test and replace
      <Tile key={`unique-key-${i}`}>
         <Anchor href={currentItem.location}>
            <img src={fixedLocation} />
         </Anchor>
      </Tile>
    });
  </Tiles>
);
```